serving/samples/helloworld-rust-nightly/Dockerfile

```diff
@@ -0,0 +1,69 @@
+# ---------- Build Container ----------
+# This uses our prebuilt dependencies
+# container as a base to speed up builds.
+# It is used to build our app.
+FROM docker.io/{username}/helloworld-rust-nightly:deps AS build
```

Comment:
Would you be able to use a publicly available image?

Reply:
It is just the first image that the readme example has us build from the Dockerfile-deps file. So each user builds their own version before building the application container, and they can then choose when to update it.

Reply:
This adds a lot of complexity to the example IMO. Why cannot the sample use the official Rust base image?

Reply:
Because to run on a scratch container, it must be built with rust nightly with the musl target, and must link against some C libraries which are themselves built with the musl target. These C libraries are not available on the official Rust base image and neither is rust nightly. Furthermore, compiling the dependencies takes a while so it is really nice to just build them once-in-a-while and have any changes to the actual application build really quickly.

Reply:
I fear that this might scare users. Hello world are supposed to convey simplicity, I am in favor of using the simplest Dockerfile and sample.

Maybe create a new sample instead of replacing the Hello World? Something like `rest-api-rust`?

Reply:
Sure, I could move it to a different folder and change the name, and then I could even add an example to the source code showing how to handle a couple different URL routes, including with params, with Rocket. I think moving forward though, users need to learn about multi-stage builds right from the beginning, because otherwise everyone is pushing 300 MB+ images into production for no reason, which also leaves open a larger attack surface.

Reply:
I agree. let's create a new sample, but update the helloworld-rust sample to mention it as a beast practice example.

Reply:
Okay, I will close this pull request and open a new one later today when I have the new example ready.